packaging/test_rule_engine_plugin_metadata_guard.py

```diff
@@ -223,6 +223,49 @@ def test_plugin_honors_admin_only_config_option_when_rodsusers_manipulate_metada
         # Clean up.
         self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
 
+    def test_plugin_disallows_rmw(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
+        self.rods.assert_icommand(['imeta', 'set', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+        try:
+            with lib.file_backed_up(config.server_config_path):
+                self.enable_rule_engine_plugin(config)
+
+                # neither user should be able to use rmw
+                self.user.assert_icommand(['imeta', 'rmw', '-C', self.user.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
+                self.rods.assert_icommand(['imeta', 'rmw', '-C', self.rods.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
+        finally:
+            # Clean up.
+            self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+    def test_plugin_cannot_be_bypassed_with_addw(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
+        self.rods.assert_icommand(['imeta', 'set', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+        try:
+            with lib.file_backed_up(config.server_config_path):
+                self.enable_rule_engine_plugin(config)
+
+                self.user.assert_icommand(['imeta', 'addw', '-C', self.user.session_collection, 'irods::addw', 'v'], 'STDERR', ['CAT_INSUFFICIENT_PRIVILEGE_LEVEL'])
```

Comment:
rodsadmin users shouldn't be able to use `addw` either.

Reply:
right, because we're disallowing the subcommand itself.

Reply:
For deprecation reasons? Currently, the regular logic blocks `addw` for non-authorized users (but will allow admins or editors). Should I make `addw` also trigger SYS_NOT_ALLOWED?

Reply:
i see what you're saying...  hmm, it's not removed yet....   just deprecated so far...

Reply:
Currently, `addw` isn't a "hole" because the regular logic still blocks it accordingly (admins can, regular users can't, editors can, etc.). I can change the code so it "fully" blocks it too (like `rmw`) if needed.

Reply:
right - so there's no hole to plug today... i think we leave it as is until we actually remove addw completely as a subcommand.

Reply:
That's true. Letting it ride!

Carry on sir.